sphinx_automodapi/autodoc_enhancements.py

```diff
@@ -58,7 +60,19 @@ def type_object_attrgetter(obj, attr, *defargs):
                 return base.__dict__[attr]
             break
 
-    return getattr(obj, attr, *defargs)
+    try:
+        return getattr(obj, attr, *defargs)
+    except AttributeError as e:
+        # for dataclasses, get the attribute from the __dataclass_fields__
+        if dataclasses.is_dataclass(obj):
+            if attr in obj.__dataclass_fields__:
+                return obj.__dataclass_fields__[attr].name
+            else:
+                # raise original AttributeError
+                raise e
+        else:
+            # raise original AttributeError
+            raise e
```

Comment:
The logic here could be simplified a bit and all put into one conditional and a raise outside of it.